class Solution {
    int diameter = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        height(root);
        return diameter;
    }
    public int height(TreeNode root){

        if(root==null){
            return 0;
        }
        int leftheight=height(root.left);
        int rightheight=height(root.right);

        diameter=Math.max(diameter,leftheight+rightheight);

        return 1+Math.max(leftheight,rightheight);
    }
}